servicetalk-log4j2-mdc-utils/build.gradle

```diff
@@ -24,8 +24,6 @@ dependencies {
   implementation project(":servicetalk-annotations")
   implementation project(":servicetalk-context-api")
 
-  runtimeOnly "org.apache.logging.log4j:log4j-core"
```

Comment:
I'm not sure why this dependency existed. It was moved from implemented to runtimeOnly [here](https://github.com/apple/servicetalk/commit/e87371c51849f4a8dea2120b08d9f45129079ff0#diff-cbcf92d0027ceb5270d1e1862a00e0b65490bc1bbc5d520cf661fe07c989b30cL27-R27).

Reply:
Our best guess is this might have become no longer needed once we changed to the SPI provider.